engine/src/main/java/org/terasology/engine/ComponentSystemManager.java

```diff
@@ -84,7 +84,14 @@ public void loadSystems(ModuleEnvironment environment, NetworkMode netMode) {
             Name moduleId = environment.getModuleProviding(type);
             RegisterSystem registerInfo = type.getAnnotation(RegisterSystem.class);
             if (registerInfo.value().isValidFor(netMode, isHeadless)) {
-                systemsByModule.put(moduleId, type);
+                if(registerInfo.requiresOptional().length != 0){
+                    for(String s : registerInfo.requiresOptional()){
+                        if(environment.get(new Name(s))!=null)systemsByModule.put(moduleId, type);
+                    }
```

Comment:
Hi @flo , I'm sorry that I have made a stupid mistake. Now I've modified a little, it runs in multiple optional dependencies tests. 


( It may be not a good concept is that because it may make dependency relations unclear ? but I think this modification may help in the case where the module integration is a 1-liner in one system, as mentioned in #2767.)